src/ui/backend.rs

```diff
@@ -11,6 +11,65 @@ use crate::{
     validator::ErrorMessage,
 };
 
+macro_rules! __print_impl {
+    (
+        $self:ident,
+        $val_expr:expr,
+        $write_fn:ident,
+        $val_ident:ident => $val_content:expr,
+        $line_ident:ident => $converter:expr
+    ) => {{
+        let $val_ident = $val_expr;
+        let __content = $val_content;
+        let __converted = newline_converter::unix2dos(&__content);
+
+        for $line_ident in __converted.split_inclusive("\r\n") {
+            if $self.is_new_line {
```

Comment:
I was coming up with a way to remove the additional flag and managed to reduce a little bit of the code at 43da960892af061394b7e5f3fcb37ff9b7aefade, wdyt?